app/helpers/encryption_helper.rb

```diff
@@ -0,0 +1,148 @@
+require "csv"
+module EncryptionHelper
+  def assign_secret_key
+    primes = []
+
+    CSV.foreach("primes_test.csv") do |row|
+      primes << row.first.to_i
+    end
+
+    length = primes.length - 1
+
+    key = [primes[rand(length)], primes[rand(length)]]
+
+    while key[0] == key[1]
+      key = [primes[rand(length)], primes[rand(length)]]
+    end
+
+    key
+  end
+
+  def euclidean_algorithm(a, b)
+    euclidean_algorithm = [[a, a / b, b, a % b]]
+    n = 0
+    while euclidean_algorithm[n][3] > 0
+      iteration = [euclidean_algorithm[n][2],
+                   euclidean_algorithm[n][2] / euclidean_algorithm[n][3],
+                   euclidean_algorithm[n][3],
+                   euclidean_algorithm[n][2] % euclidean_algorithm[n][3]]
+      euclidean_algorithm << iteration
+
+      n += 1
+    end
+    euclidean_algorithm
+  end
+
+  def gcd(a, b)
+    e_a = euclidean_algorithm(a, b)
+    if e_a.length > 1
+      e_a[e_a.length - 2][3]
+    else
+      e_a[0][2]
+    end
+  end
+
+  def calculate_public_key(p, q)
+    m = p * q
+    phi_of_m = (p - 1) * (q - 1)
+
+    k = rand(10000000)
+
+    while gcd(phi_of_m, k) != 1
+      k = rand(10000000)
+    end
+
+    [m, k]
+  end
+
+  def encoder_hash
+    { "a" => "11", "b" => '12', "c" => "13", "d" => "14", "e" => "15",
+      "f" => "16", "g" => '17', "h" => "18", "i" => "19", "j" => "20",
+      "k" => "21", "l" => '22', "m" => "23", "n" => "24", "o" => "25",
+      "p" => "26", "q" => '27', "r" => "28", "s" => "29", "t" => "30",
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.